pkg/cloud-init/cloud-init.go

```diff
@@ -570,7 +570,43 @@ func GenerateLocalData(vmiName string, namespace string, data *CloudInitData) er
 	return nil
 }
 
-// Lists all vmis cloud-init has local data for
-func listVmWithLocalData() ([]*v1.VirtualMachineInstance, error) {
-	return diskutils.ListVmWithEphemeralDisk(cloudInitLocalDir)
+// Lists all vmis ephemeral disk has local data for
+func listVmWithLocalData(localPath string) ([]*v1.VirtualMachineInstance, error) {
```

Comment:
actually, I think you may have found some code that is effectively dead and can be removed.

From what I can tell, this code is only called by a unit test and the unit test doesn't exercise anything except this code. 

This function likely served a meaningful purpose in the past, but after refactors became unnecessary.

Reply:
We cannot delete `listVmWithLocalData` as it is used by a test. It should be somewhere. I propose to put it in the production code, as the implementation of obj.read or obj.list should always be parallel to obs.write. It is far more important to maintain proper code abstraction layers than excluding a function from runtime.

If you agree @davidvossel , please approve.

Reply:
>We cannot delete listVmWithLocalData as it is used by a test.

ha, right. take a look at that test that exercises this function. i think the test is literally just testing this functions output... and that's the only place this logic is ever executed.

so, why does this function exist still? I could be wrong, but it appears to serve no purpose, and the unit test that exercises it appears to be something left over after a refactor.

Reply:
oh dear; I somehow convinced myself that the test was creating real local data. I see that the user of `cloudinit.ListVmWithLocalData` was dropped back in commit 58b636ad28ed2d63db92f2ecc0f16541e41c29eb

I'll just drop the entire thing.